onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -389,6 +389,10 @@ class PlannerImpl {
 
   // Find if freelist contains a buffer of the same size as output_arg
   bool FindReusableTensor(const onnxruntime::NodeArg& output_arg, OrtValueIndex* reusable_tensor) {
+    if(!context_.GetEnableMemoryReuse()) {
+      return false;
+    }
```

Comment:
This feature is quite useful by itself. Could you create a separate PR for it to be merged in first?